assets/js/components/ClusterDetails/ConnectionSettings.jsx

```diff
@@ -0,0 +1,195 @@
+      {
+        title: 'Connection User',
+        key: 'user',
+        render: (content, item) => {
+          return (
+            <input
+              type="text"
+              id={item.host_id}
+              className="rounded-lg border-transparent flex-1 appearance-none border border-gray-100 w-full py-2 px-4 bg-white text-gray-700 placeholder-gray-400 shadow-sm text-base focus:outline-none focus:ring-1 focus:focus:border-transparent"
+              placeholder="Provide a user if different from the default one"
+              value={content || ''}
+              onChange={({ target: { value } }) =>
+                setHostConnectionUser(item.host_id, value)
+              }
+            />
+          );
+        },
+      },
+      {
+        title: 'Default User',
+        key: 'default_user',
+        render: (content, item) => {
+          return (
+            <Switch.Group as="div" className="flex items-center">
+              <Switch.Label className="mr-4">{content}</Switch.Label>
+              <Switch
+                checked={item.isDefaultUser}
+                className={classNames(
+                  item.isDefaultUser ? 'bg-jungle-green-500' : 'bg-gray-200',
+                  'relative inline-flex flex-shrink-0 h-6 w-11 border-2 border-transparent rounded-full cursor-pointer focus:outline-none transition-colors ease-in-out duration-200'
+                )}
+                onChange={() => toggleUseDefaultConnectionUser(item.host_id)}
+              >
+                <span
+                  aria-hidden="true"
+                  className={classNames(
+                    item.isDefaultUser ? 'translate-x-5' : 'translate-x-0',
+                    'inline-block h-5 w-5 rounded-full bg-white shadow transform ring-0 transition ease-in-out duration-200'
+                  )}
+                />
+              </Switch>
+            </Switch.Group>
+          );
+        },
+      },
+    ],
+  };
+
+  return (
+    <div>
+      <Table config={connectionSettingsTableConfig} data={localSettings} />
+      <div className="place-items-end flex">
+        <button
+          disabled={saving}
+          className="flex justify-center items-center bg-jungle-green-500 hover:opacity-75 text-white font-bold py-2 px-4 rounded"
+          onClick={() => {
+            const ls = localSettings.map((hostSettings) => {
+              if (hostSettings.isDefaultUser) {
+                return { ...hostSettings, user: null };
+              }
+              return hostSettings;
+            });
+
+            dispatch({
+              type: 'SAVE_CLUSTER_CONNECTION_SETTINGS',
+              payload: {
+                cluster: cluster?.id,
+                settings: ls,
+              },
+            });
+            // dispatch({
+            //     type: 'CHECKS_SELECTED',
+            //     payload: { checks: selectedChecks, clusterID: clusterID },
+            // });
+            // navigate(`/clusters/${clusterID}/checks/results`);
```

Comment:
:eyes: could we strip this?

Reply:
Next, need for the next iteration! :sweat_smile: